sbt launcher terminates BSP server when sbt is not on path

When starting the sbt launcher from an environment that does not have the sbt script in the path, and no sbt server is running, the launcher script simply terminates after printing `[info] server was not detected. starting an instance`. This is the case when starting from IntelliJ IDEA.

## steps

1. start IntelliJ with Scala plugin from the UI (on Mac?)
2. import an sbt project configured to 1.4.x as BSP, choose "sbt" and not "sbt with Bloop" if prompted

alternatively if that doesn't repro:
1. start IntelliJ from a script, but set PATH to not include the sbt script
2. as above

## problem

1. the import will time out
2. in the build toolwindow, we see only
```[info] server was not detected. starting an instance```

In the background, what happens is that the sbt launcher simply terminates if it doesn't find the sbt launcher on the path.

## expectation

* the server should be able to be started from the same JVM used to start the launcher
* the sbt script should not be necessary at all, as sbt may not even have been installed on the system

Oho there is some progress...
I have Scala plugin version 2020.3.21 installed.

`reee@Krunoslavs-MacBook-Pro exercises % /Users/reee/Library/Java/JavaVirtualMachines/openjdk-15.0.2/Contents/Home/bin/java -Xms100m -Xmx100m -classpath /Users/reee/Library/Application Support/JetBrains/IntelliJIdea2020.3/plugins/Scala/launcher/sbt-launch.jar xsbt.boot.Boot -bsp --sbt-launch-jar=/Users/reee/Library/Application Support/JetBrains/IntelliJIdea2020.3/plugins/Scala/launcher/sbt-launch.jar
Error: Could not find or load main class Support.JetBrains.IntelliJIdea2020.3.plugins.Scala.launcher.sbt-launch.jar
Caused by: java.lang.ClassNotFoundException: Support.JetBrains.IntelliJIdea2020.3.plugins.Scala.launcher.sbt-launch.jar`

seems that `Application Support` folder wasn't escaped properly so

`reee@Krunoslavs-MacBook-Pro exercises % /Users/reee/Library/Java/JavaVirtualMachines/openjdk-15.0.2/Contents/Home/bin/java -Xms100m -Xmx100m -classpath /Users/reee/Library/"Application Support"/JetBrains/IntelliJIdea2020.3/plugins/Scala/launcher/sbt-launch.jar xsbt.boot.Boot -bsp --sbt-launch-jar=/Users/reee/Library/"Application Support"/JetBrains/IntelliJIdea2020.3/plugins/Scala/launcher/sbt-launch.jar`

is producing just:
`[info] server was not detected. starting an instance`

and exits

--- a/protocol/src/main/scala/sbt/internal/bsp/BuildServerConnection.scala
+++ b/protocol/src/main/scala/sbt/internal/bsp/BuildServerConnection.scala
@@ -28,6 +28,7 @@ object BuildServerConnection {
     val sbtLaunchJar = classPath
       .split(File.pathSeparator)
       .find(jar => SbtLaunchJar.findFirstIn(jar).nonEmpty)
+      .map(_.replaceAllLiterally(" ", "%20"))
     val argv =
       Vector(
         s"$javaHome/bin/java",

